src/main/java/hudson/plugins/promoted_builds/PromotionCause.java

```diff
@@ -53,4 +53,40 @@ public void print(final TaskListener listener) {
         );
     }
     // }}}
+
+    @Override
+    public int hashCode() {
+        int hash = 7;
```

Comment:
minor: i'm not familiar with how you make a hash function good enough. adding comments would be great.